ckan/lib/search/jobs.py

```diff
@@ -0,0 +1,70 @@
+# encoding: utf-8
+
+# (canada fork only): background search index rebuilding
+#TODO: upstream contrib!!
+
+import json
+import datetime
+from rq import get_current_job
+
+import ckan.logic as logic
+import ckan.lib.search as search
+import ckan.model as model
+
+from ckan.plugins import toolkit
+
+from logging import getLogger
+log = getLogger(__name__)
+
+
+def reindex(package_ids=None, group_id=None):
+    context = {
+        'model': model,
+        'ignore_auth': True,
+        'validate': False,
+        'use_cache': False
+    }
+
+    _entity_id = group_id if group_id else toolkit.config.get('ckan.site_id')
+    task = {
+        'entity_id': _entity_id,
+        'entity_type': 'group' if group_id else 'site',
+        'task_type': 'search_rebuild',
+        'last_updated': str(datetime.datetime.now(datetime.timezone.utc)),
+        'state': 'running',
+        'key': 'search_rebuild',
+        'value': '{}',
+        'error': '{}',
+    }
+
+    try:
+        task = logic.get_action('task_status_show')(context, {'entity_id': _entity_id,
+                                                              'task_type': 'search_rebuild',
+                                                              'key': 'search_rebuild'})
+        task['state'] = 'running'
+        task['last_updated'] = str(datetime.datetime.now(datetime.timezone.utc))
+        logic.get_action('task_status_update')({'session': model.meta.create_local_session(), 'ignore_auth': True}, task)
+    except logic.NotFound:
+        pass
+
+    value = json.loads(task.get('value', '{}'))
+    error = json.loads(task.get('error', '{}'))
+
+    value['job_id'] = get_current_job().id
+
+    for _dataset_id, _total, _indexed, _error in search.rebuild(force=True, package_ids=package_ids):
```

Comment:
for submitting to core the `_` prefixes don't follow the normal coding standards (in Python this means either "this variable is not used anywhere" or "this is intended as a private member" for object attributes)

Reply:
will do that change in upstream PR

Reply:
I lied I did it now.